cromwell_tools/utilities.py

```diff
@@ -232,3 +238,91 @@ def validate_cromwell_label(label_object):
 
     if err_msg != '':
         raise ValueError(err_msg)
+
+
+def prepare_workflow_manifest(wdl_file, inputs_file=None, options_file=None, dependencies=None, label_file=None,
+                              collection_name=None, on_hold=False, **inputs_file_n):
+    """Prepare the submission manifest for a workflow submission.
+
+    Args:
+        wdl_file (Union[str, io.BytesIO]): The workflow source file to submit for execution. Could be either the path
+            to the file (str) or the file content in io.BytesIO.
+        inputs_file (Optional[Union[str, io.BytesIO]]): The input data in JSON format. Could be either the path to the
+            file (str) or the file content in io.BytesIO. (default None)
+        options_file (Optional[Union[str, io.BytesIO]]): The Cromwell options file for workflows. Could be either
+        the path to the file (str) or the file content in io.BytesIO. (default None)
+        dependencies (Optional[Union[str, List[str], io.BytesIO]]): Workflow dependency files. Could be the path to
+            the zipped file (str) containing dependencies, a list of paths(List[str]) to all dependency files to be
+            zipped or a zipped file in io.BytesIO. (default None)
+        label_file(Optional[Union[str, _io.BytesIO]]): JSON file containing a collection of key/value pairs for workflow
+            labels. (default None)
+        collection_name (Optional[str]): Collection in SAM that the workflow should belong to, if use CaaS. (
+            default None)
+        on_hold: (Optional[bool]) Whether to submit the workflow in "On Hold" status. (default False)
+        **inputs_file_n: Keyword arguments for up to many inputs_file that could be accepted by Cromwell. Each of them
+            should have a type of Optional[Union[str, io.BytesIO]]. Any arguments that doesn't follow the
+            "inputs_file_n" naming pattern will be ignored.
+
+    Returns
+        workflow_manifest (dict): A dictionary representing the workflow manifest ready for workflow submission.
+    """
+    workflow_manifest = {}
+
+    # Compose WDL source file
+    workflow_manifest['workflowSource'] = _download_to_BytesIO_if_string(wdl_file)
+
+    # Compose WDL inputs1
+    if inputs_file:
+        workflow_manifest['workflowInputs'] = _download_to_BytesIO_if_string(inputs_file)
+
+    # Compose other WDL inputs
+    for idx, (input_file_key, input_file) in enumerate(inputs_file_n.items()):
+        if re.fullmatch('inputs\_file\_[0-9]+', input_file_key):
```

Comment:
Is this magic worth it? If Cromwell accepted an arbitrary number of inputs I could see it, but (at least from [the Swagger](https://cromwell.caas-prod.broadinstitute.org/swagger/index.html?url=/swagger/cromiam.yaml#/Workflows/post_api_workflows__version_)) it seems capped at 5, which wouldn't be that bad to make explicit.

Reply:
Good point! So it's not listed on the Swagger of CaaS as you pointed out, but described on normal Cromwell instance's [Swagger](https://cromwell.mint-dev.broadinstitute.org/swagger/index.html?url=/swagger/cromwell.yaml#/Workflows/submit), which said "Submits a workflow to Cromwell. Note that this endpoint can accept an unlimited number of input files via workflowInputs_N but swagger needs them to be explicitly defined so we have provided 5 as an example.". So I'm leaning towards using this trick instead of listing all of the inputs. 

I think there are a lot of small gaps/differences between CaaS and Cromwell we need to be aware of, such as the `labelor` feature. I'm still not sure how we could better harmonize them in cromwell-tools (maybe a probe to check whether using CaaS or Cromwell, or a user option to be passed in). What do you think?

Reply:
A new idea for the inputs JSONs: instead of taking both `inputs_file` and `**inputs_file_n`, this method could take `inputs_files: (List[Union[str, io.BytesIO]])`. Users can still pass in as many JSONs as they want, but they don't have to worry about matching the magic naming convention. Something like:

```python
if inputs_files:
    for idx, input_file in enumerate(inputs_files):
        if idx == 0:
            key_name = 'workflowInputs'
        else:
            key_name = 'workflowInputs_{X}'.format(X=idx+1)

    workflow_manifest[key_name] = _download_to_BytesIO_if_string(input_file)
```

Reply:
Unsure about the best way to handle API skew between CaaS / Cromwell / older versions of Cromwell, but agreed it's not the best UX to expose a bunch of options that might not actually do anything... probably worth a separate issue.

Reply:
I like the idea of setting the users free from the restrictive naming convention, and the snippet above seems pretty great except that forcing user input `List[]` make the signature of the `/submit` method farther from the Cromwell endpoint. Given that most of the workflows just use 1 inputs_file, what do you think if we make 1 addition to the above snippet:
```python
if inputs_files:
    if not isinstance(inputs_files, list):
        inputs_files = [inputs_files]
    for idx, input_file in enumerate(inputs_files):
        if idx == 0:
            key_name = 'workflowInputs'
        else:
            key_name = 'workflowInputs_{X}'.format(X=idx+1)
```